### PR TITLE
Ask specific translations

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Update the frontend to only request the translations that it needs, instead of loading all translations of all possible integrations that will be encountered.

Requires https://github.com/home-assistant/core/pull/34329

Also adds icons to onboarding

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/state/translations-mixin.ts [hunk 1]
```diff
@@ -75,12 +83,41 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
...
+        // indicates if category is loaded
+        category in this.__loadedTranslations &&
+        // if this.__loadedTranslations[category] is true, it was loaded with config flow
+        (!configFlow || this.__loadedTranslations[category])
+      ) {
+        return;
+      }
+
+      if (!integration) {
+        this.__loadedTranslations[category] = configFlow === true;
```
Comment (bramkragten) [inline src/state/translations-mixin.ts:111]: Can we make this clearer and more future proof?
```
        this.__loadedTranslations[category] = { configFlow };
```

File: src/state/translations-mixin.ts [hunk 2]
```diff
@@ -75,12 +83,41 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
...
+    ) {
+      // We don't cache individual integrations
+      // Our cache stores
+      if (
+        !force &&
+        !integration &&
+        // indicates if category is loaded
+        category in this.__loadedTranslations &&
+        // if this.__loadedTranslations[category] is true, it was loaded with config flow
+        (!configFlow || this.__loadedTranslations[category])
```
Comment (bramkragten) [inline src/state/translations-mixin.ts:105]:
```
        (!configFlow || this.__loadedTranslations[category].configFlow)
```

File: src/state/translations-mixin.ts (hunk 1 above)
Comment (bramkragten) [reply src/state/translations-mixin.ts:111]: Also, if we will fetch the `title` translation of one page with, and the other without `configFlow` we keep busting the cache?

File: build-scripts/webpack.js [hunk 3]
```diff
@@ -70,6 +70,7 @@ const createWebpackConfig = ({
         __BUILD__: JSON.stringify(latestBuild ? "latest" : "es5"),
         __VERSION__: JSON.stringify(version),
         __DEMO__: false,
+        __BACKWARDS_COMPAT__: false,
```
Comment (balloob) [inline build-scripts/webpack.js:73]: This new constant allows us to add some code paths to remain backwards compatibility. This path will be removed if backwards compat is turned off. 

Currently only Cast will have this turned on.

**Pull request merged**